### PR TITLE
Added a functional React component example in react.md docs

### PR DESCRIPTION
**Pull request by hcbd**
## Description
The current React example for Video JS only describes a class component.
I added a working example of a React functional component with a fix for the dispose() method removing the elements from the DOM. 

## Specific Changes proposed
Added an example and implementation example for a React functional component (React v16.8+).

Added to the top: 

## React Functional Component Example
```jsx
import React from "react";
import videojs from "video.js";
import "video.js/dist/video-js.css";

export const VideoJS = ( props ) => {

  const videoRef = React.useRef(null);
  const { options } = props;

  // This seperate functional component fixes the removal of the videoelement 
  // from the DOM when calling the dispose() method on a player
  const VideoHtml = ( props ) => (
    <div data-vjs-player>
      <video ref={videoRef} className="video-js vjs-big-play-centered" />
    </div>
  );

  React.useEffect( () => {
    const videoElement = videoRef.current;
    let player;
    if( videoElement ) {
      player = videojs( videoElement, options, () => {
        console.log("player is ready");
      });
    }
    return () => {
      if( player ) {
        player.dispose();
      }
    }
  }, [options]);

  return (<VideoHtml />);
}
export default VideoJS;
```

__Implementation Example:__

```jsx
import React from "react";
import VideoJS from './VideoJS' // point to where the functional component is stored

const App = () => {

  const videoJsOptions = { // lookup the options in the docs for more options
    autoplay: true,
    controls: true,
    responsive: true,
    fluid: true,
    sources: [{
      src: '/path/to/video.mp4',
      type: 'video/mp4'
    }]
  }
  
  return (
    <>
      <div>Rest of app here</div>
      
      <VideoJS options={videoJsOptions}/>
      
      <div>Rest of app here</div>
    </>
  );
}
```
## React Class Component Example

.....[ rest of document, added the css file import to the class example ].....
...
.

## Requirements Checklist
- [ ] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [ ] Change has been verified in an actual browser (Chrome, Firefox, IE)
  - [ ] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: docs/guides/react.md [hunk 1]
```diff
@@ -2,11 +2,91 @@
 
 Here's a basic ReactJS player implementation.
 
+## React Functional Component Example
```
Comment (gkatsev) [inline docs/guides/react.md:5]:
```suggestion
## React Functional Component and useEffect Example
```

File: docs/guides/react.md [hunk 2]
```diff
@@ -2,11 +2,91 @@
...
+    }
+  }, [options]);
+
+  return (<VideoHtml />);
+}
+export default VideoJS;
+
+```
+
+__Implementation Example:__
```
Comment (gkatsev) [inline docs/guides/react.md:46]:
```suggestion
You can then use it like this: (see [options guide][options] for option information)
```

File: docs/guides/react.md [hunk 3]
```diff
@@ -2,11 +2,91 @@
...
+    </>
+  );
+}
+
+```
+
+
+--------
+
+
```
Comment (gkatsev) [inline docs/guides/react.md:80]:
```suggestion
```

File: docs/guides/react.md [hunk 4]
```diff
@@ -2,11 +2,91 @@
 
 Here's a basic ReactJS player implementation.
```
Comment (gkatsev) [inline docs/guides/react.md:3]:
```suggestion
Here are a couple ReactJS player implementations.
```

**Review by gkatsev — COMMENTED** [review]
Hey, sorry it's taken so long to take a look at this. A couple of super minor changes.
Really appreciate it, thanks!

**Comment by hcbd** [comment]
> Hey, sorry it's taken so long to take a look at this. A couple of super minor changes.
> Really appreciate it, thanks!

No problem, and fixing my bad English is always appreciated.

**Review by gkatsev — APPROVED** [review]

**Review by gkatsev — APPROVED** [review]

**Pull request merged**